src/python/zquantum/core/circuit/_gate.py

```diff
@@ -84,11 +84,16 @@ def __eq__(self, another_gate):
     def __repr__(self):
         return f"zquantum.core.circuit.Gate(name={self.name}, qubits={self.qubits}, params={self.params})"
 
-    def create_evaluated_gate(self, symbols_map):
+    def evaluate(self, symbols_map):
         """
-        Creates a copy of gate with symbolic params
+        Returns a copy of a gate with specified symbolic parameters evaluated to provided values.
+
+        Args:
+            symbols_map list(tuple(sympy.Basic, number)): List containing symbols and values that they should take.
         """
-        evaluated_gate = copy.deepcopy(self)
+        gate_class = type(self)
```

Comment:
What is this line for?

Reply:
Instead of making a copy (in particular a deep copy, which copies all the referenced objects etc.), we create a new object, which is of the same class as the initial one (and not Gate class, in case you ever have an inherited class).

Reply:
Yes, but you don't use the `gate_class` variable after you assign it

Reply:
I do:
<img width="524" alt="Screen Shot 2020-06-12 at 10 48 00 AM" src="https://user-images.githubusercontent.com/7314136/84515235-30aff380-ac9a-11ea-9d1d-fafc8e079826.png">